src/features/campaigns/components/ActivityList/index.tsx

```diff
@@ -17,12 +17,61 @@ import useClusteredActivities, {
   CLUSTER_TYPE,
 } from 'features/campaigns/hooks/useClusteredActivities';
 import CanvassAssignmentListItem from './items/CanvassAssignmentListItem';
+import ActivityListItem, { STATUS_COLORS } from './items/ActivityListItem';
 
 interface ActivitiesProps {
   activities: CampaignActivity[];
   orgId: number;
 }
 
+interface LazyActivitiesBoxProps extends BoxProps {
+  index: number;
+}
+
+const LazyActivitiesBox = ({
+  index,
+  children,
+  ...props
+}: LazyActivitiesBoxProps) => {
+  const [inView, setInView] = useState(false);
+  const boxRef = useRef<HTMLElement>();
+
+  useEffect(() => {
+    if (!boxRef.current) {
+      return;
+    }
+
+    const observer = new IntersectionObserver((entries) => {
+      entries.forEach((entry) => {
+        if (entry.isIntersecting) {
+          setInView(true);
+        }
+      });
+    });
+
+    observer.observe(boxRef.current);
```

Comment:
Do we need to unobserve or destroy the observer when the component unmounts, or will that be handled automatically by the browser when React removes the element from the DOM?

Reply:
From what I can find in discussions the intent is for garbage collection to deal with it if observer is no longer observing any elements (using weak references):
[https://github.com/w3c/csswg-drafts/issues/6397](https://github.com/w3c/csswg-drafts/issues/6397)